src/codelabs/iterables/index.md

```diff
@@ -0,0 +1,571 @@
+</iframe>
+
+{{site.alert.secondary}}
+  **Quick review:**
+  * Filter the elements of an `Iterable` with `where`.
+  * The output of `where` is another `Iterable`.
+  * Use `takeWhile` and `skipWhile` to obtain elements until or after a condition is met.
+  * The output of these methods can be an empty `Iterable` too.
+{{site.alert.end}} 
+
+## Mapping
+
+Mapping `Iterables` with the method `map` allows you to apply a function
+over each one of the elements, replacing each element by a new one.
+
+{% prettify dart %}
+  Iterable<int> output = numbers.map((number) => number * 10);
+{% endprettify %}
+
+In this example, each element of the `Iterable` numbers is multiplied by 10.
+
+You can also use `map` to transform an element into a different object, for example,
+to convert all `int` to `String` as you can see in the example below.
+
+{% prettify dart %}
+  Iterable<String> output = numbers.map((number) => number.toString());
+{% endprettify %}
+
+{{site.alert.note}}
+  `map` returns a "lazy" `Iterable`, meaning that the supplied function will
+  only be called when the elements are iterated.
+{{site.alert.end}}
+
+### Example: Using map to change elements
+
+Run this example to see how `map` can be used to multiply all the elements
+of an `Iterable` by two. What do you think the output will be?
+
+[//]: https://gist.github.com/ba4352e121b19ecb1b171cce553de72f
+<iframe
+  src="{{site.dartpad-embed}}?id=ba4352e121b19ecb1b171cce553de72f"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="200"
+  width="100%" >
+</iframe>
+
+### Exercise: Mapping to a different type
+
+In the previous example you multiplied the elements of an `Iterable` by two.
+Both the input and the output of that operation were an `Iterable` of `int`.
+
+In this exercise, you code takes an `Iterable` of `User`, and you need to
+return an `Iterable` that contains the `age` multiplied by two.
+
+As a reminder, the class `User` contains a member `age` of the type `int`.
+
+{% prettify dart %}
+  class User {
+    String name;
+    int age;
+  }
+{% endprettify %}
+
+[//]: https://gist.github.com/c2ca9085b2376471b30569d570590894
+<iframe
+src="{{site.dartpad-embed}}?id=c2ca9085b2376471b30569d570590894&theme=dark"
+frameborder="no"
+height="325"
+width="100%" >
+</iframe>
+
+{{site.alert.secondary}}
+  **Quick review:**
+  * `map` applies a function to all the elements of an `Iterable`.
+  * The output of `map` is another `Iterable`.
+  * The function is not evaluated until the `Iterable` is iterated.
+{{site.alert.end}} 
+
+## Exercise: Putting it all together
+
+It’s time to practice what you’ve learned in one final exercise.
+
+This exercise provides the class `Email` with a constructor, that takes a `String`.
+
+{% prettify dart %}
+class Email {
+  String address;
+
+  Email(
+    this.address,
+  );
+}
+{% endprettify %}
+
+This exercise also provides the function `checkIfEmailInvalid` that returns `true` if
```

Comment:
It's generally frowned upon to have negative tests (https://dart.dev/guides/language/effective-dart/design#prefer-the-positive-name-for-a-boolean-property-or-variable), plus "Email" is generally an adjective or short for an email message; it's confusing to use it to refer to an email address. Also, booleans should generally use is/has/can/... verbs, for code readability (https://dart.dev/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable). So maybe:

`checkIfEmailInvalid` -> `!isValidEmailAddress` (global)
(or `isValidAddress` if space is tight)

And maybe rename `Email` to `EmailAddress` or `Address` everywhere?

I really did find the term "Email" confusing at first, or else I wouldn't make such a stink about it. :)

Reply:
+1 on the naming `isValidEmailAddress`. If we do need the negative name I'd go with `isInvalidEmailAddress`.

Reply:
thanks a lot for the review, I have redone the final exercise with your suggestions